Verify Pipelines is installed before installing PaC

guillaumerose opened the pull request:
If Pipelines is not installed, kubectl apply will fail for all Triggers ressources.

guillaumerose committed:
Verify Tekton is installed before installing PaC

If Tekton is not installed, kubectl apply will fail for
all Triggers ressources.

chmouel commented on pkg/cmd/tknpac/bootstrap/bootstrap.go:
```diff
@@ -60,6 +61,13 @@ func install(ctx context.Context, run *params.Run, opts *bootstrapOpts) error {
+		return err
+	}
+	if !tektonInstalled {
+		return errors.New("tekton API not found on the cluster. Please install OpenShift Pipelines first")
```
we want to make it compatible tekton upstream too, so maybe if we can change the wording to Tekton :)

guillaumerose committed:
Close body only if there is no error

It avoids a panic when a real error occurs:
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x2081b5d]

goroutine 1 [running]:
github.com/openshift-pipelines/pipelines-as-code/pkg/cmd/tknpac/bootstrap.detectSelfSignedCertificate({0x269d090, 0xc000126008}, {0xc000050780, 0x5f})
	/gomod/pipelines-as-code/pkg/cmd/tknpac/bootstrap/route.go:57 +0xbd

chmouel commented:
/lgtm
